### DISCUSSION TITLE
Can I use utils and runtime in the browser?

### DISCUSSION THREAD

**Discussion by phrozen** [panva/jose#184] [Q&A]
Hi,

Can I use the internal tools and runtime as a library (in browser), specifically base64url encode/decode and SHA-256 Digest. I figured I'm already using jose and there is no need to either come up with my own implementation or add an extra dependency.

**Comment by panva** [comment]
utils yes, that's public api, runtime is not part of the public api

**Reply by phrozen** [reply]
Thanks! I guess I can use `crypto.Subtle` to digest to SHA256 easily enough then.

How can I import the base64 functions in here: https://github.com/panva/jose/blob/main/src/util/base64url.ts

`import { Base64UrlEncode, Base64UrlDecode } from 'jose/util/base64url';`

This does not seem to work in my React app, I did install only the browser runtime, does that make a difference to the import paths?

**Comment by panva** [comment]
```js
import * as base64url from 'jose/util/base64url';
```

**Reply by phrozen** [reply]
I think something might be wrong in my setup, cause I get:

`Module not found: Can't resolve 'jose/util/base64url'`

Thanks, I will look into it.
